vm.go

```diff
@@ -365,20 +365,20 @@ func floatToValue(f float64) (result Value) {
 	return valueFloat(f)
 }
 
-func assertInt64(v Value) (int64, bool) {
+func assertInt64(v Value) (int64, bool, bool) {
```

Comment:
I'm open to other ideas on how to surface whether or not the value is a valueInt64 but I figured this approach lets us avoid additional casts.

Reply:
Yes good idea!

Reply:
hmm this is a tricky one. I also don't think this is particularly intuitive. Have you considered using generics to perform the increment/decrement operations? That might allow you to avoid these casts completely and be more performant than the current approach. e.g.
```golang
type numericValues interface {
	valueInt | valueUInt32 | valueInt32 | valueInt64 | valueFloat
}

func increment[T numericValues](n T) T {
	return n + 1
}
```

Reply:
The generic approach is really performant 😲 

```
goja (realm)$ go test -ldflags="-r /Users/calvin.nix/dev/baas/etc/dylib/lib" --bench="BenchmarkEmptyLoop" --run="BenchmarkEmptyLoop" -benchtime 5s
goos: darwin
goarch: arm64
pkg: github.com/dop251/goja
BenchmarkEmptyLoop-10            1245040              4804 ns/op
```

this benchmark was using this logic:

```
func (_inc) exec(vm *vm) {
	v := vm.stack[vm.sp-1]

	switch i := v.(type) {
	case valueInt:
		v = increment(i)
	case valueInt64:
		v = increment(i)
	case valueFloat:
		v = increment(i)
	default:
		panic("ah")
	}

	vm.stack[vm.sp-1] = v
	vm.pc++
}
```